drivers/media/platform/raspberrypi/rp1_cfe/pisp_fe.c

```diff
@@ -433,26 +433,63 @@ static int pisp_fe_pad_set_fmt(struct v4l2_subdev *sd,
 
 	switch (format->pad) {
 	case FE_STREAM_PAD:
-	case FE_OUTPUT0_PAD:
-	case FE_OUTPUT1_PAD:
 		cfe_fmt = find_format_by_code(format->format.code);
 		if (!cfe_fmt || !(cfe_fmt->flags & CFE_FORMAT_FLAG_FE_OUT))
 			cfe_fmt = find_format_by_code(MEDIA_BUS_FMT_SRGGB16_1X16);
 
 		format->format.code = cfe_fmt->code;
+		format->format.field = V4L2_FIELD_NONE;
 
-		break;
+		fmt = v4l2_subdev_get_pad_format(sd, state, FE_STREAM_PAD);
+		*fmt = format->format;
 
-	case FE_STATS_PAD:
-	case FE_CONFIG_PAD:
-		format->format.code = MEDIA_BUS_FMT_FIXED;
-		break;
-	}
+		fmt = v4l2_subdev_get_pad_format(sd, state, FE_OUTPUT0_PAD);
+		*fmt = format->format;
 
-	fmt = v4l2_subdev_get_pad_format(sd, state, format->pad);
-	*fmt = format->format;
+		fmt = v4l2_subdev_get_pad_format(sd, state, FE_OUTPUT1_PAD);
+		*fmt = format->format;
```

Comment:
Does this mean that userland *has* to set the FE_STREAM_PAD format before FE_OUTPUT0_PAD and FE_OUTPUT1_PAD?

Reply:
Yes. You should always configure the sink side of the subdev first, and then the source side.

Of course, once you have configured the sink side, you now what the sink side format is, and you can then configure the source side multiple times (presuming all the source side configs are compatible with the sink side).

I think you've asked about this, or similar cases, earlier too. Do you see some issues with this? Afaic, this is how all well-behaved subdev drivers work. See Documentation/userspace-api/media/v4l/dev-subdev.rst, search for "propagate".

Reply:
Ok sounds reasonable.